pkg/asset/machines/azure/machines.go

```diff
@@ -78,12 +79,26 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 	return machines, nil
 }
 
-func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string, userDataSecret string, clusterID string, role string, azIdx *int, hyperVGen string) (*machineapi.AzureMachineProviderSpec, error) {
+func provider(platform *azure.Platform, mpool *azure.MachinePool, osImage string, userDataSecret string, clusterID string, role string, azIdx *int, capabilities map[string]string) (*machineapi.AzureMachineProviderSpec, error) {
 	var az *string
 	if len(mpool.Zones) > 0 && azIdx != nil {
 		az = &mpool.Zones[*azIdx]
 	}
 
+	hyperVGen, err := getHyperVGenerationVersion(capabilities)
+	if err != nil {
+		return nil, err
+	}
+
+	if mpool.VMNetworkingType == "" {
+		acceleratedNetworking, err := getVMNetworkingCapability(capabilities)
+		if err != nil {
+			return nil, err
+		}
+		if acceleratedNetworking {
+			mpool.VMNetworkingType = string(azure.VMnetworkingTypeAccelerated)
+		}
```

Comment:
Let's do an `else` here with a logrus `INFO` that the selected instance type does not support accelerated networking.